simple_certmanager/admin.py

```diff
@@ -92,6 +94,51 @@ class SigningRequestAdmin(admin.ModelAdmin):
         ),
     )
 
+    def get_urls(self):
+        urls = super().get_urls()
+        my_urls = [
+            path(
+                "dowload_csr/<int:pk>",
+                self.admin_site.admin_view(self.download_csr),
+                name="download_csr",
+            )
+        ]
+        return my_urls + urls
+
+    def download_csr(self, request, pk):
+        # Here I don't use the change permission because it won't be given after
+        # the CSR is signed and the certificate is uploaded.
+        if self.has_add_permission(request):
```

Comment:
```suggestion
        if self.has_view_permission(request):
```

It's what we're doing.